scheduler/src/cook/mesos/util.clj

```diff
@@ -298,65 +298,59 @@
-          expanded-end (Date. (+ (.getTime end)
-                                 (-> 1 t/hours t/in-millis)))
-          jobs
-          (->> (d/seek-datoms db :avet :job/user user (d/entid-at db :db.part/user expanded-start))
-               (take-while #(and (< (.e %) (d/entid-at db :db.part/user expanded-end))
-                                 (= (.a %) (d/entid db :job/user))
-                                 (= (.v %) user)))
-               (map #(.e %))
-               (map (partial d/entity db))
-               (filter #(<= (.getTime start) (.getTime (:job/submit-time %))))
-               (filter #(< (.getTime (:job/submit-time %)) (.getTime end)))
-               (filter #(= :job.state/completed (:job/state %)))
-               (filter #(not (:job/custom-executor %))))]
-      (->>
-        (cond->> jobs
-                 (instance-states state) (filter #(= state (job-ent->state %)))
-                 name-filter-fn (filter #(name-filter-fn (:job/name %))))
-        (take limit)))))
-
-;; This is a separate query from completed jobs because most running/waiting jobs
-;; will be at the end of the time range, so the query is somewhat inefficent.
-;; The standard datomic query performs reasonably well on the smaller set of
-;; running and waiting jobs, so it's implemented that way to keep things simple.
+            entid-start (d/entid-at db :db.part/user expanded-start)
+            entid-end (d/entid-at db :db.part/user expanded-end)
+            job-user-entid (d/entid db :job/user)]
+           (->> (d/seek-datoms db :avet :job/user user entid-start)
+                (take-while #(and (< (:e %) entid-end)
+                                  (= (:a %) job-user-entid)
+                                  (= (:v %) user)))
+                (map #(:e %))
+                (map (partial d/entity db))
+                (filter #(<= (.getTime start) (.getTime (:job/submit-time %))))
+                (filter #(< (.getTime (:job/submit-time %)) (.getTime end)))
+                (filter #(= state-keyword (:job/state %)))
+                (filter #(not (:job/custom-executor %))))))
+
+;; This differs from the get-active-jobs-by-user-and-state is also looking up based on task state.
+(defn get-completed-jobs-by-user
+      "Returns all completed job entities for a particular user
+       in the specified timeframe, without a custom executor. Supports looking up based
+       on task state 'success' and 'failed' if passed into 'state'"
+      [db user start end limit state name-filter-fn]
+      (timers/time!
+        get-completed-jobs-by-user-duration
+        (->>
+          (cond->> (get-jobs-by-user-and-state-and-submit db user start end :job.state/completed)
+                   (instance-states state) (filter #(= state (job-ent->state %)))
+                   name-filter-fn (filter #(name-filter-fn (:job/name %))))
+          (take limit))))
+
+;; This query looks for all jobs by job state only (i.e., no 'success' or 'failed')
```

Comment:
Seems like this is appropriate for the docstring

Reply:
I viewed that as supplementary material. This is an helper function, not intended for external use.

Reply:
Changing.